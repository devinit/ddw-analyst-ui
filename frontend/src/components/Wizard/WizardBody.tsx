import React, { FunctionComponent } from 'react';
import { Card, Tab } from 'react-bootstrap';

const WizardBody: FunctionComponent = ({ children }) => {
  return (
    <Card.Body>
      <Tab.Content>{children}</Tab.Content>
    </Card.Body>
  );
};

export { WizardBody };
