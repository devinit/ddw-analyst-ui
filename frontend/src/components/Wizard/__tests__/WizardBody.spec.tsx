/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardBody } from '../WizardBody';

describe('WizardBody', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardBody />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardBody>
        <div>My Child</div>
      </WizardBody>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
