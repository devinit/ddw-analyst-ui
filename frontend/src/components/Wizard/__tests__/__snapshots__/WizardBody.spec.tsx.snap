// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardBody renders the default structure correctly 1`] = `
<div
  className="card-body"
>
  <div
    className="tab-content"
  />
</div>
`;

exports[`WizardBody renders with children 1`] = `
<div
  className="card-body"
>
  <div
    className="tab-content"
  >
    <div>
      My Child
    </div>
  </div>
</div>
`;
